src/scrollspy.ts

```diff
@@ -246,20 +259,25 @@ export class ScrollSpy extends Component<ScrollSpyOptions> {
   }
 
   _enter() {
-    ScrollSpy._visibleElements = ScrollSpy._visibleElements.filter(value => value.getBoundingClientRect().height !== 0);
+    ScrollSpy._visibleElements = ScrollSpy._visibleElements.filter(
+      (value) => value.getBoundingClientRect().height !== 0
+    );
 
     if (ScrollSpy._visibleElements[0]) {
-      const actElem = document.querySelector(this.options.getActiveElement(ScrollSpy._visibleElements[0].id));
+      const actElem = document.querySelector(
+        this.options.getActiveElement(ScrollSpy._visibleElements[0].id)
+      );
       actElem?.classList.remove(this.options.activeClass);
 
-      if (ScrollSpy._visibleElements[0].M_ScrollSpy && this.id < ScrollSpy._visibleElements[0].M_ScrollSpy.id) {
+      if (
+        ScrollSpy._visibleElements[0]['M_ScrollSpy'] &&
+        this.id < ScrollSpy._visibleElements[0]['M_ScrollSpy'].id
+      ) {
```

Comment:
I think we might need to revert this, since it introduces backwards compatibility issues with applications using the previous object orientated approach, propose to just set it to warnings until we can declare the correct type definition